src/main/java/com/example/onlinebookstore/service/category/CategoryServiceImpl.java

```diff
@@ -0,0 +1,54 @@
+package com.example.onlinebookstore.service.category;
+
+import com.example.onlinebookstore.dto.category.CategoryDto;
+import com.example.onlinebookstore.dto.category.CreateCategoryRequestDto;
+import com.example.onlinebookstore.exception.EntityNotFoundException;
+import com.example.onlinebookstore.mapper.CategoryMapper;
+import com.example.onlinebookstore.model.Category;
+import com.example.onlinebookstore.repository.CategoryRepository;
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class CategoryServiceImpl implements CategoryService {
+    private final CategoryRepository categoryRepository;
+    private final CategoryMapper categoryMapper;
+
+    @Override
+    public List<CategoryDto> findAll(Pageable pageable) {
+        return categoryRepository.findAll(pageable).stream()
+                .map(categoryMapper::toDto)
+                .toList();
+    }
+
+    @Override
+    public CategoryDto getById(Long id) {
+        Category category = categoryRepository.findById(id)
+                .orElseThrow(() -> new EntityNotFoundException("Cannot find category by id " + id)
+        );
+        return categoryMapper.toDto(category);
```

Comment:
you can simplify  it using the method map of the Optional class

Reply:
i changed implementation to map, but is this realy more simple for reading ?